instagram-app/src/authentication/withAuthenticate.jsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import LoginPage from "../components/Login/LoginPage";
+
+const withAuthenticate = Component => {
+  return class extends React.Component {
```

Comment:
Good job with this HOC, the only suggestion that I would made is to pass the LoginPage component as a prop. This way your component can achieve true re usability.